pandas/tests/frame/test_sorting.py

```diff
@@ -227,6 +228,30 @@ def test_stable_descending_multicolumn_sort(self):
                                    kind='mergesort')
         assert_frame_equal(sorted_df, expected)
 
+    def test_sort_multi_index(self):
+        # GH 25775
+        s = ',a,b,c,d,e' + \
+            '0,0.548,7,0.602,0.544,0.423' + \
+            '1,0.645,4,0.891,0.963,0.383' + \
+            '2,0.791,5,0.568,0.925,0.071' + \
+            '3,0.087,0,0.832,0.778,0.870' + \
+            '4,0.978,7,0.461,0.780,0.118' + \
+            '5,0.639,1,0.944,0.521,0.414' + \
+            '6,0.264,7,0.456,0.568,0.018' + \
+            '7,0.617,6,0.616,0.943,0.681' + \
+            '8,0.359,4,0.697,0.060,0.666' + \
```

Comment:
is there a reason we are using a fixed string here rather than a generated frame with a seed?

Reply:
The other reviewer requested "a single deterministic example" in contrast to iterating through a range of seeds. I wasn't sure what they exactly meant by "deterministic" i.e. whether fixing a seed is good enough or should the dataframe be explicitly given

Reply:
Yeah, I meant a single explicit `DataFrame` that minimally reproduces the error.  Setting a random seed and the generating random floats had nothing to do with the actual error in question, so IMO it made it less obvious as to what's going on.  It also resulted in there be a lot more data than necessary to reproduce the error, which makes it harder to follow what's being tested.

So basically something like:
```python
df = DataFrame({'a': [3, 1, 2], 'b': [0, 0, 0],
                'c': [0, 1, 2], 'd': list('abc')})
result = df.set_index(list('abc')).sort_index(level=list('ba'))

expected = DataFrame({'a': [1, 2, 3], 'b': [0, 0, 0],
                      'c': [1, 2, 0], 'd': list('bca')})
expected = expected.set_index(list('abc'))

tm.assert_frame_equal(result, expected)
```

Note that `df` as defined above does indeed not sort correctly on 0.23.4:
```python
In [1]: import numpy as np; import pandas as pd; pd.__version__
Out[1]: '0.23.4'

In [2]: df = pd.DataFrame({'a' : [3, 1, 2], 'b': [0, 0, 0,], 'c': [0, 1, 2], 'd': list('abc')})

In [3]: df.set_index(list('abc')).sort_index(level=list('ba'))
Out[3]: 
       d
a b c   
3 0 0  a
1 0 1  b
2 0 2  c
```
I suppose you could parametrize over `level` if you wanted to, as adding `'c'` still results in invalid sorting:
```python
In [4]: df.set_index(list('abc')).sort_index(level=list('bac'))
Out[4]: 
       d
a b c   
3 0 0  a
1 0 1  b
2 0 2  c
```

Reply:
Thanks @jschendel, switched to your simpler example.